data/modules/CargoRun/CargoRun.lua

```diff
@@ -343,7 +343,8 @@ local onChat = function (form, ref, option)
 		form:SetMessage(howmuch)
 
 	elseif option == 3 then
-		if Game.player.freeCapacity < ad.amount then
+		if (not ad.pickup and Game.player.freeCapacity < ad.amount) or
+			(ad.pickup and Game.player.totalCargo < ad.amount) then
```

Comment:
I don't know how to deal with #3450. Besides that I have messed up something with this PR.